src/input/target/steam_deck_uhid.rs

```diff
@@ -0,0 +1,851 @@
+                    0x2d,
+                    0x01,
+                    0x05,
+                    0x12,
+                    0x00,
+                    0x00,
+                    0x02,
+                    0x00,
+                    0x00,
+                    0x00,
+                    0x00,
+                    0x0a,
+                    0x2b,
+                    0x12,
+                    0xa9,
+                    0x62,
+                    0x04,
+                    0xad,
+                    0xf1,
+                    0xe4,
+                    0x65,
+                    0x09,
+                    0x2e,
+                    0x00,
+                    0x00,
+                    0x00,
+                    0x0b,
+                    0xa0,
+                    0x0f,
+                    0x00,
+                    0x00,
+                    0x0d,
+                    0x00,
+                    0x00,
+                    0x00,
+                    0x00,
+                    0x0c,
+                    0x00,
+                    0x00,
+                    0x00,
+                    0x00,
+                    0x0e,
+                    0x00,
+                    0x00,
+                    0x00,
+                    0x00,
+                    0x00,
+                    0x00,
+                    0x00,
+                    0x00,
+                    0x00,
+                    0x00,
+                    0x00,
+                    0x00,
+                    0x00,
+                    0x00,
+                    0x00,
+                    0x00,
+                    0x00,
+                    0x00,
+                    0x00,
+                    0x00,
+                ];
+                data.to_vec()
+            }
+            ReportType::GetSerial => {
+                // Reply with the serial number
+                // [ReportType::GetSerial, 0x14, 0x01, ..serial?]?
+                log::debug!("Sending serial number: {}", self.serial_number);
+                let mut data = vec![0x0, ReportType::GetSerial as u8, 0x14, 0x01];
+                let mut serial_data = self.serial_number.as_bytes().to_vec();
+                data.append(&mut serial_data);
+                data.resize(64, 0);
+                data
+            }
+            // Don't care about other types
+            _ => vec![],
+        };
+
+        // Write the report reply to the HIDRAW device
+        if let Err(e) = device.write_get_report_reply(id, 0, data) {
+            log::warn!("Failed to write get report reply: {:?}", e);
+            return Err(e.to_string().into());
+        }
+
+        Ok(())
+    }
+
+    fn handle_set_report(
+        &mut self,
+        id: u32,
+        _report_number: u8,
+        _report_type: uhid_virt::ReportType,
+        data: Vec<u8>,
+    ) -> Result<Vec<OutputEvent>, Box<dyn Error>> {
+        let Some(report_id) = data.get(1) else {
+            return Ok(vec![]);
+        };
+        self.current_report = (*report_id).try_into()?;
```

Comment:
```suggestion
        self.current_report = match (*report_id).try_into() {
            Ok(id) => id,
            Err(err) => {
                log::warn("Unknown report id: {report_id}");
                return Ok(());
            }
        };
```

Reply:
I did something similar, but this had some issues.